### ISSUE TITLE
Bad index in CsMatBase::iter

### ISSUE THREAD

**Issue by avangogo** [vbarrielle/sprs#129]
The following code
```
let mut matrix = sprs::CsMat::zero( (3,100) );
matrix.insert(2, 3, 42);
for (v, (i, j)) in matrix.iter() {
  println!("index = ({}, {}), value = {}", i, j, v)
}
```
produces the output
> index = (1, 3), value = 42

while the expected index would be (2, 3)

**Comment by vbarrielle** [comment]
Thanks for the report, I'll investigate and fix this.

**Comment by vbarrielle** [comment]
As it turns out, the issue was in the `iter()` method which would improperly handle adjacent empty rows. That's fixed in #130.

On a side note, beware of constructing sparse matrices by inserting, it can be really slow.
